effekt/shared/src/main/scala/effekt/generator/llvm/Transformer.scala

```diff
@@ -273,6 +283,16 @@ object Transformer {
         emit(FAdd(name, ConstantDouble(x), ConstantDouble(0)));
         transform(rest)
 
+      case machine.LiteralUTF8String(v@machine.Variable(bind, _), utf8, rest) =>
+        MC.staticText += s"$bind.lit" -> utf8
```

Comment:
What do we do, if the same string shows up multiple times? Like `" "`?

Reply:
We encode each occurring literal separately; I doubt LLVM aliases them. I proposed to @phischu to build one singular (minimal) long byte array in which every occurring statically known one is a subarray (an optimization which is aided by the fact that we do not employ zero-terminated strings), yet he objected.